### PR TITLE
Support Webhook for Typetalk

### PR DESCRIPTION
**Pull request by is2ei**
https://github.com/go-gitea/gitea/issues/5496

Added Webhook support for Typetalk
Could you please review it? 😃

### CODE REVIEW

**Comment by is2ei** [comment]
I checked the test result and I think it is not caused by my changes.

https://drone.gitea.io/go-gitea/gitea/4547/8
```
+ make clean
go clean -i ./...
rm -rf gitea dist modules/{options,public,templates}/bindata.go \
	integrations*.test \
	integrations/gitea-integration-pgsql/ integrations/gitea-integration-mysql/ integrations/gitea-integration-sqlite/ integrations/gitea-integration-mssql/ \
	integrations/indexers-mysql/ integrations/indexers-pgsql integrations/indexers-sqlite integrations/indexers-mssql \
	integrations/mysql.ini integrations/pgsql.ini integrations/mssql.ini
+ make generate
package github.com/jteeuwen/go-bindata/...: github.com/jteeuwen/go-bindata/...: invalid import path: malformed import path "github.com/jteeuwen/go-bindata/...": double dot
Makefile:95: recipe for target 'generate' failed
make: *** [generate] Error 1
```

I ran the command in my local env. It didn't cause the error.

```
horie at MBP in ~/src/code.gitea.io/gitea on support-webhook-typetalk [?]
$ make generate
go generate code.gitea.io/gitea code.gitea.io/gitea/cmd code.gitea.io/gitea/models code.gitea.io/gitea/models/migrations code.gitea.io/gitea/modules/auth code.gitea.io/gitea/modules/auth/ldap code.gitea.io/gitea/modules/auth/oauth2 code.gitea.io/gitea/modules/auth/openid code.gitea.io/gitea/modules/auth/pam code.gitea.io/gitea/modules/avatar code.gitea.io/gitea/modules/base code.gitea.io/gitea/modules/cache code.gitea.io/gitea/modules/context code.gitea.io/gitea/modules/cron code.gitea.io/gitea/modules/generate code.gitea.io/gitea/modules/highlight code.gitea.io/gitea/modules/httplib code.gitea.io/gitea/modules/indexer code.gitea.io/gitea/modules/lfs code.gitea.io/gitea/modules/log code.gitea.io/gitea/modules/mailer code.gitea.io/gitea/modules/markup code.gitea.io/gitea/modules/markup/csv code.gitea.io/gitea/modules/markup/external code.gitea.io/gitea/modules/markup/markdown code.gitea.io/gitea/modules/markup/orgmode code.gitea.io/gitea/modules/metrics code.gitea.io/gitea/modules/minwinsvc code.gitea.io/gitea/modules/notification code.gitea.io/gitea/modules/notification/base code.gitea.io/gitea/modules/notification/ui code.gitea.io/gitea/modules/options code.gitea.io/gitea/modules/pprof code.gitea.io/gitea/modules/private code.gitea.io/gitea/modules/process code.gitea.io/gitea/modules/public code.gitea.io/gitea/modules/recaptcha code.gitea.io/gitea/modules/search code.gitea.io/gitea/modules/setting code.gitea.io/gitea/modules/ssh code.gitea.io/gitea/modules/sync code.gitea.io/gitea/modules/templates code.gitea.io/gitea/modules/test code.gitea.io/gitea/modules/user code.gitea.io/gitea/modules/util code.gitea.io/gitea/modules/validation code.gitea.io/gitea/routers code.gitea.io/gitea/routers/admin code.gitea.io/gitea/routers/api/v1 code.gitea.io/gitea/routers/api/v1/admin code.gitea.io/gitea/routers/api/v1/convert code.gitea.io/gitea/routers/api/v1/misc code.gitea.io/gitea/routers/api/v1/org code.gitea.io/gitea/routers/api/v1/repo code.gitea.io/gitea/routers/api/v1/swagger code.gitea.io/gitea/routers/api/v1/user code.gitea.io/gitea/routers/api/v1/utils code.gitea.io/gitea/routers/dev code.gitea.io/gitea/routers/org code.gitea.io/gitea/routers/private code.gitea.io/gitea/routers/repo code.gitea.io/gitea/routers/routes code.gitea.io/gitea/routers/user code.gitea.io/gitea/routers/user/setting code.gitea.io/gitea/routers/utils
bindata.go
bindata.go
bindata.go
```

File: options/locale/locale_de-DE.ini [hunk 1]
```diff
@@ -1115,6 +1115,7 @@ settings.slack_domain=Domain
 settings.slack_channel=Kanal
 settings.add_discord_hook_desc=<a href="%s">Discord</a>-Integration zu deinem Repository hinzufügen.
 settings.add_dingtalk_hook_desc=<a href="%s">Dingtalk</a>-Integration zu deinem Repository hinzufügen.
+settings.add_typetalk_hook_desc=<a href="%s">Typetalk</a>-Integration zu deinem Repository hinzufügen.
```
Comment (lunny) [inline options/locale/locale_de-DE.ini]: Don't change translation files except english file.

**Review by lunny — CHANGES_REQUESTED** [review]

File: options/locale/locale_de-DE.ini (hunk 1 above)
Comment (is2ei) [reply options/locale/locale_de-DE.ini]: @lunny Ok! I'll updated it.

Comment (is2ei) [reply options/locale/locale_de-DE.ini]: @lunny I updated it! could you review it again?

**Comment by lunny** [comment]
There is a bug on go1.11.3 I think that made the CI failed.

**Comment by is2ei** [comment]
> There is a bug on go1.11.3 I think that made the CI failed.

@lunny  Ok! We need to wait fot the CI to be fixed, right?

File: modules/setting/setting.go [hunk 2]
```diff
@@ -1606,7 +1606,7 @@ func newWebhookService() {
 	Webhook.QueueLength = sec.Key("QUEUE_LENGTH").MustInt(1000)
 	Webhook.DeliverTimeout = sec.Key("DELIVER_TIMEOUT").MustInt(5)
 	Webhook.SkipTLSVerify = sec.Key("SKIP_TLS_VERIFY").MustBool()
-	Webhook.Types = []string{"gitea", "gogs", "slack", "discord", "dingtalk"}
+	Webhook.Types = []string{"gitea", "gogs", "slack", "discord", "dingtalk","typetalk"}
```
Comment (adelowo) [inline modules/setting/setting.go]:
```suggestion
	Webhook.Types = []string{"gitea", "gogs", "slack", "discord", "dingtalk", "typetalk"}
```

Comment (is2ei) [reply modules/setting/setting.go]: @adelowo Thanks! I committed the change.

**Comment by is2ei** [comment]
@lunny @adelowo 
The test is passed! If it is ok, could you give me lgtm?

**Comment by lunny** [comment]
@is2ei I will try this and since v1.7 new feature is freezed, this could be merged on v1.8 .

**Comment by lunny** [comment]
Please resolve the conflicted files.

**Comment by lafriks** [comment]
Conflicts need to be resolved

**Comment by 6543** [comment]
@is2ei can you resolve conflicts?

PS: some code has moved so take care of :D

**Comment by lunny** [comment]
Please resolve the conflicts.

**Comment by 6543** [comment]
@is2ei #9422 made some changes ... so It is worth looking at it ... (you have to resolve conflicts anyway)

**Comment by 6543** [comment]
@is2ei I know this PR is old ... but do you still have passion to get it upstream?

**Pull request closed without merging**